doc/tutorial-restful-blog.md

```diff
@@ -0,0 +1,314 @@
+
+```tcl
+proc entry_create {entry_title entry_content} {
+    #| Create a new blog entry
+    set entry_id [db_seq entry_id_seq]
+    db_dml "insert into entries\
+           [sql_insert entry_id entry_title entry_content]"
+    return $entry_id
+}
+```
+
+Save, restart, and try submitting a blog with "test" for the title and content. You should now get "Not Found" again. However, you should see the address is now `localhost/entries/1`, unlike before. Go into psql and run `SELECT * FROM entries;` and you will see that your blog has in fact been successfully submitted.
+
+The reason you are seeing "Not Found" is because you have been redirected to the address that should show you the details of your new blog entry, but the server has not yet been given instructions to deal with that request. To fix that, add this code to your `url_handlers`:
+
+```tcl
+register GET /entries/:entry_id {entry_id} {
+    #| View an entry
+    return [entry_get $entry_id]
+}
+```
+
+You should now be getting "Internal Server Error" once again. Check the logs if you like, or just look at the code we just added - we are trying use an `entry_get` proc that we haven't written yet. Add the following to `entry.tcl`:
+
+```tcl
+proc entry_get {entry_id} {
+    #| Return html summary for this entry 
+    db_1row {
+	select
+	entry_title,
+	entry_content
+	from
+	entries
+	where entry_id=:entry_id
+    }
+    set html ""
+    append html [h h1 $entry_title]
+    append html [h div $entry_content]
+    return $html
+}
+```
+
+After saving and restarting, try submitting a blog post. You should, at last, be able to see the post you have submitted (and you should be able to see your previous posts if you change your address to have their entry_id instead). You have successfully added basic create and read functionality to your site. 
+
+## register, h, and form
+
+Before moving on, let's review some of the procs we have been using in our code.
+
+We have made use of the `register` proc throughout the `url_handlers.tcl` file. This is used to register a path, so that the server has instructions in place for how to deal with, for example, a request such as `GET entries/1`. Note the used of colon variables, for example in your handler for viewing an entry:
+
+```tcl
+register GET /entries/:entry_id {entry_id} {
+    #| View an entry
+    return [entry_get $entry_id]
+}
+```
+
+By putting a colon at the start of `:entry_id`, we instruct the handler to treat it as a variable name and pass its value to `entry_id` to be used in the function body. For more detail on the register proc, [see its documentation](registration.md).
+
```

Comment:
I suggest: For more detail, see [documentation on the register proc](registration.md)

It is preferable that  hyperlink context provided stands on its own.

"see its documentation" requires the context of the sentence it is in.

Reply:
Alright - I'll make a similar change for the other links